src/rdb/rdb_raft.c

```diff
@@ -1556,6 +1557,144 @@ rdb_gc_yield(void *arg)
 	return 0;
 }
 
+static void
+rdb_raft_checkpoint_wait(void *arg, uint64_t wait_id, uint64_t *commit_id)
+{
+	struct rdb              *db    = arg;
+	struct rdb_chkpt_record *dcr   = &db->d_chkpt_record;
+	struct umem_store       *store = dcr->dcr_store;
+
+	if (store->stor_ops->so_wal_id_cmp(store, dcr->dcr_commit_id, wait_id) >= 0)
+		goto out;
+
+	ABT_mutex_lock(db->d_raft_mutex);
+	dcr->dcr_waiting = 1;
+	dcr->dcr_wait_id = wait_id;
+	do {
+		sched_cond_wait(db->d_commit_cv, db->d_raft_mutex);
```

Comment:
It's better to call ABT_cond_wait() directly instead of sched_cond_wait() here, since the tx is supposed to be committed very soon.
sched_cond_wait() is supposed to be called when the waiter expects a long term waiting (like the waiter in rdb_checkpointd()), so that the scheduler is able to tell that we have such waiters not sensitive to latency, and enter relax mode despite of such waiters.